### ISSUE TITLE
[Enhancement] Enable use of sections in declarative list

### ISSUE THREAD

**Issue by RasmusKjeldgaard** [kirbydesign/designsystem#1816]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
I would like to be able to use a kirby-item-group or similar, to declaratively group related items together. 

Together with #1910 it should provide us with the necessary structure to replicate the current look and feel of the list, once we tackle the styling of these in #1814 and #1817. 

## Describe the solution you'd like
I would expect to be able to do something like 
```
<kirby-list>
    <kirby-list-section>
        <kirby-item-divider>
            ...
        </kirby-item-divider>

        <kirby-list-item>
            ...
        </kirby-list-item>
    <kirby-list-section>
</kirby-list>
```

This would mean that Kirby provides a thin wrapper around the `ion-item-group` that is currently used within the `kirby-list` so it can be provided directly instead.  

## Have you considered any alternatives?
.

## Are there any additional context?
See #1794 and #1813 for context.

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

**Comment by RasmusKjeldgaard** [comment]
Split this into two (see #1910), and put some of the story points on that.

**Comment by RasmusKjeldgaard** [comment]
As of the current Ionic version (v5), the [ion-item-group](https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/item-group/item-group.tsx) offers no additional styling, and only adds a `role="group"` to itself. So for now, we have decided to implement this ourselves, to avoid an unnecessarily bloated HTML structure when using this component.
